diff --git a/Lib/asyncio/unix_events.py b/Lib/asyncio/unix_events.py
index 1ff8c427da..0e8eb719ed 100644
--- a/Lib/asyncio/unix_events.py
+++ b/Lib/asyncio/unix_events.py
@@ -369,6 +369,11 @@ class _UnixSelectorEventLoop(selector_events.BaseSelectorEventLoop):
                 fut.set_result(total_sent)
                 return
 
+        # On 32-bit architectures truncate to 1GiB to avoid OverflowError,
+        # see bpo-38319.
+        if sys.maxsize < 2 ** 32:
+            blocksize = min(blocksize, 2 ** 30)
+
         try:
             sent = os.sendfile(fd, fileno, offset, blocksize)
         except (BlockingIOError, InterruptedError):