From 6dbbd125f941a9f1d6ebd39bfddadf44521cd0e9 Mon Sep 17 00:00:00 2001
From: Marc Mueller <30130371+cdce8p@users.noreply.github.com>
Date: Sat, 16 Aug 2025 16:00:31 +0200
Subject: [PATCH] gh-90548: Fix musl version detection with --strip-all

https://github.com/python/cpython/pull/137864


diff --git a/Lib/platform.py b/Lib/platform.py
index da15bb4717b..5dbe7f0bcd7 100644
--- a/Lib/platform.py
+++ b/Lib/platform.py
@@ -194,6 +194,7 @@ def libc_ver(executable=None, lib='', version='', chunksize=16384):
         | (GLIBC_([0-9.]+))
         | (libc(_\w+)?\.so(?:\.(\d[0-9.]*))?)
         | (musl-([0-9.]+))
+        | (libc.musl(?:-\w+)?.so(?:\.(\d[0-9.]*))?)
         """,
         re.ASCII | re.VERBOSE)
 
@@ -219,7 +220,10 @@ def libc_ver(executable=None, lib='', version='', chunksize=16384):
                     continue
                 if not m:
                     break
-            libcinit, glibc, glibcversion, so, threads, soversion, musl, muslversion = [
+            (
+                libcinit, glibc, glibcversion, so, threads, soversion,
+                musl, muslversion, musl_so, musl_sover
+            ) = [
                 s.decode('latin1') if s is not None else s
                 for s in m.groups()]
             if libcinit and not lib:
@@ -241,6 +245,10 @@ def libc_ver(executable=None, lib='', version='', chunksize=16384):
                 lib = 'musl'
                 if not ver or V(muslversion) > V(ver):
                     ver = muslversion
+            elif musl_so:
+                lib = 'musl'
+                if musl_sover and (not ver or V(musl_sover) > V(ver)):
+                    ver = musl_sover
             pos = m.end()
     return lib, version if ver is None else ver
 
